pkgs/development/python-modules/w1thermsensor/default.nix

```diff
@@ -0,0 +1,68 @@
+{ lib
+, buildPythonPackage
+, fetchPypi
+, pkgs
+, setuptools
+, aiofiles
+, click
+, coverage
+, tomli
+, pytest
+, pytest-mock
+, pytest-asyncio
+, pytestCheckHook
+, pythonOlder
+}:
+buildPythonPackage rec {
+  pname = "w1thermsensor";
+  version = "2.0.0";
+  format = "pyproject";
+
+  src = fetchPypi {
+    inherit pname version;
+    hash = "sha256-EcaEr4B8icbwZu2Ty3z8AAgglf74iZ5BLpLnSOZC2cE=";
+  };
+
+  postPatch = ''
+    sed -i 's/3\.5\.\*/3.5/' setup.py
+  '';
+
+  nativeBuildInputs = [
+    setuptools
+  ];
+
+  propagatedBuildInputs = [
+    aiofiles
+    click
+  ];
+
+  # Don't try to load the kernel module in tests.
+  env.W1THERMSENSOR_NO_KERNEL_MODULE = 1;
+
+  nativeCheckInputs = [
+    pytest-mock
+    pytest-asyncio
+  ] ++ lib.optionals (pythonOlder "3.11") [
+    tomli
+    # Tests for 2.0.0 currently fail on python3.11
+    # https://github.com/timofurrer/w1thermsensor/issues/116
```

Comment:
Would be better to set `doCheck = pythonOlder "3.11";` instead.

Reply:
If I do that, I would lose the import test on 3.11, no? It seems preferable to me to keep that, since it's still passing.

Reply:
Ah, I didn't realize that they were independent of `doCheck`. Thanks.  I've updated the PR to use `doCheck` as you suggested.